tests/containers_test.go

```diff
@@ -64,14 +100,51 @@ func TestPulumiDockerImage(t *testing.T) {
 				e.DeleteEnvironment()
 			}()
 
-			stackName := fmt.Sprintf("%s/container-%s-%x", stackOwner, template, time.Now().UnixNano())
-			e.RunCommand("pulumi", "new", template, "-y", "-f", "-s", stackName)
+			stackName := fmt.Sprintf("%s/container-%s-%x", stackOwner, test.template, time.Now().UnixNano())
+			e.RunCommand("pulumi", "new", test.template, "-y", "-f", "-s", stackName)
 
 			example := base.With(integration.ProgramTestOptions{
-				Dir: e.RootPath,
+				Dir:    e.RootPath,
+				Config: test.config,
 			})
 
 			integration.ProgramTest(t, &example)
 		})
 	}
 }
+
+func TestCLIToolTests(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Azure CLI", func(t *testing.T) {
+		t.Parallel()
+
+		clientId := mustEnv(t, "ARM_CLIENT_ID")
+		clientSecret := mustEnv(t, "ARM_CLIENT_SECRET")
+		tenantId := mustEnv(t, "ARM_TENANT_ID")
+		subscriptionId := mustEnv(t, "ARM_SUBSCRIPTION_ID")
+
+		cmd := exec.Command("az", "login", "--service-principal",
+			"--username", clientId,
+			"--password", clientSecret,
+			"--tenant", tenantId)
+		_, err := cmd.Output()
+		require.NoError(t, err)
+
+		cmd = exec.Command("az", "account", "show")
+		out, err := cmd.Output()
+		require.NoError(t, err)
+		result := map[string]interface{}{}
+		json.Unmarshal(out, &result)
+		require.Equal(t, subscriptionId, result["id"])
+	})
+}
+
+func mustEnv(t *testing.T, env string) string {
+	t.Helper()
+	v := os.Getenv(env)
+	if v == "" {
+		t.Fatalf("Required environment variable %q not set", env)
```

Comment:
Should these be skipped, so tests can be run locally?  Or I guess that maybe doesn't make a lot of sense for the docker containers?  It's a pattern we used for other cloud tests.

Reply:
I was wondering about that, and I worried about a test not running because of a broken CI setup. I'd rather it fail loudly. 

I was thinking that for running tests locally I could add something that runs `esc run pulumi/providers.all -- test_command_here`. That gives you easy access to all the env vars.

Reply:
Yeah, this definitely makes sense during development.

The `esc run` is also great, though I think it only works for pulumi employees (unless I'm misunderstanding esc).  

Kindof an aside, but I'd love to get to the point where we skip no tests in CI, so we can assert that there are no skipped tests during the CI run, but sprinkle `t.Skip()` more liberally through tests that don't necessarily work locally.

Either way, I'm happy to get what you have here now merged.